tests/internal/test_module.py

```diff
@@ -0,0 +1,158 @@
+from contextlib import contextmanager
```

Comment:
we should add subprocess tests here as well to test normal `import module` and `from module import thing`

we also probably want to test importing child modules e.g. you want to know when `parent` is imported, but you do `import parent.child.grandchild.greatgrandchild`, do you get notified of all of them?

Reply:
👍  for the subprocess tests, although this has been tested extensively elsewhere already 🙂 . And yes, I believe you get notified for all the modules that are actually imported during the `import bla`